lib/CMakeLists.txt

```diff
@@ -201,44 +212,76 @@ target_compile_definitions(quda_cpp PRIVATE $<TARGET_PROPERTY:quda,COMPILE_DEFIN
-           $<$<CONFIG:STRICT>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_STRICT}>
-           $<$<CONFIG:RELEASE>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_RELEASE}>
-           $<$<CONFIG:DEBUG>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_DEBUG}>
-           $<$<CONFIG:HOSTDEBUG>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_HOSTDEBUG}>
-           $<$<CONFIG:DEVICEDEBUG>:-Xcompiler
-           ${CMAKE_CXX_FLAGS_DEVICEDEBUG}>
-           >)
+
+  # Pick the right set of flags Apparently I cannot do this with generator expressions
+  if(CMAKE_BUILD_TYPE STREQUAL "DEVEL")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_DEVEL}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "STRICT")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_STRICT}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "RELEASE")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_RELEASE}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_DEBUG}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "HOSTDEBUG")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_HOSTDEBUG}")
+  elseif(CMAKE_BUILD_TYPE STREQUAL "DEVICEDEBUG")
+    set(PROPAGATED_FLAGS "${CMAKE_CXX_FLAGS_DEVICEDEBUG}")
+  endif()
+
+  # Turne the flags into a CMAKE list
+  string(REPLACE " " ";" PROPAGATED_FLAG_LIST ${PROPAGATED_FLAGS})
+
+  foreach(FLAG ${PROPAGATED_FLAG_LIST})
+    target_compile_options(quda PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>: -Xcompiler=${FLAG}>)
+  endforeach()
 endif()
 
+# Specific comnfig dependent warning suppressions and lineinfo forwarding
+target_compile_options(
+  quda
+  PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:
+          $<$<CONFIG:DEVEL>:-lineinfo
+          -Xcompiler
+          -Wno-unknown-pragmas,-Wno-unused-function,-Wno-unused-local-typedef,-Wno-unused-private-field>
+          $<$<CONFIG:STRICT>:
+          -Xcompiler
+          -Wno-unknown-pragmas,-Wno-unused-function,-Wno-unused-local-typedef,-Wno-unused-private-field>
+          $<$<CONFIG:HOSTDEBUG>:-lineinfo>
+          $<$<CONFIG:SANITIZE>:-lineinfo
+          -Xcompiler
+          -fsanitize=address,-fsanitize=undefined>>
+          $<$<COMPILE_LANG_AND_ID:CUDA,Clang>:
+          $<$<CONFIG:DEVEL>:-Wno-unknown-pragmas
+          -Wno-unused-function
+          -Wno-unused-local-typedef
+          -Wno-unused-private-field>
+          $<$<CONFIG:STRICT>:-Wno-unknown-pragmas
+          -Wno-unused-function
+          -Wno-unused-local-typedef
+          -Wno-unused-private-field>
+          $<$<CONFIG:HOSTDEBUG>:>
+          $<$<CONFIG:SANITIZE>:-fsanitize=address,-fsanitize=undefined>
+          >)
+
 # some clang warnings should be warning even when turning warnings into errors
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   target_compile_options(quda_cpp PUBLIC $<$<COMPILE_LANGUAGE:CXX>:-Wno-error=unused-private-field
                                          -Wno-error=unused-function>)
-  target_compile_options(quda PRIVATE $<$<COMPILE_LANGUAGE:CUDA>: "SHELL:-Xcompiler -Wno-error=unused-private-field"
```

Comment:
Did this get moved somewhere else for `CUDA` builds?

Reply:
Can't seem to find it anywhere, based on searching for `SHELL:-Xcompiler -Wno-error=unused-private-field`

Reply:
Added a clarifying question below.

Reply:
I think this is now taken care of by this line:

```target_compile_options(
  quda
  PRIVATE $<$<COMPILE_LANG_AND_ID:CUDA,NVIDIA>:
          $<$<CONFIG:DEVEL>:-lineinfo
          -Xcompiler
          -Wno-unknown-pragmas,-Wno-unused-function,-Wno-unused-local-typedef,-Wno-unused-private-field>
          $<$<CONFIG:STRICT>:
          -Xcompiler
          -Wno-unknown-pragmas,-Wno-unused-function,-Wno-unused-local-typedef,-Wno-unused-private-field>
          $<$<CONFIG:HOSTDEBUG>:-lineinfo>
          $<$<CONFIG:SANITIZE>:-lineinfo
          -Xcompiler
          -fsanitize=address,-fsanitize=undefined>>
          $<$<COMPILE_LANG_AND_ID:CUDA,Clang>:
          $<$<CONFIG:DEVEL>:-Wno-unknown-pragmas
          -Wno-unused-function
          -Wno-unused-local-typedef
          -Wno-unused-private-field>
          $<$<CONFIG:STRICT>:-Wno-unknown-pragmas
          -Wno-unused-function
          -Wno-unused-local-typedef
          -Wno-unused-private-field>
          $<$<CONFIG:HOSTDEBUG>:>
          $<$<CONFIG:SANITIZE>:-fsanitize=address,-fsanitize=undefined>```

for cases when the CUDA compiler is NVCC - instead of `-Wno-error=unused-private-field` the shorter `-Wno-unused-private-field` is used. When the CUDA compiler is Clang, there is no need for propagating these as they are already on the regular CXXFLAGS...